core/src/main/scala/cats/data/NonEmptyList.scala

```diff
@@ -0,0 +1,159 @@
+package cats
+package data
+
+import cats.std.list._
+//it needs a Functor
+//it needs semigroup - combine, filter
+
+/**
+ *  A data type which represents a single element (head) and some other
+ * structure (tail).
+ */
+final case class NonEmptyList[A](head: A, tail: List[A] = List[A]()) {
```

Comment:
I would prefer not to have a default parameter here -- we can support calling with one argument via an additional `apply` method on the companion object.